### DISCUSSION TITLE
TypeScript, Git/GitHub

### DISCUSSION THREAD

**Discussion by CodeRider101** [SE-TINF22B2/G2-Schoolify#61] [Technologien]
Warum benutzen wir TypeScript, Git/GitHub und nicht andere Technologien auf dieser Ebene?
@tilwbr

**Comment by tilwbr** [comment]
### Erklärung Typescript
TypeScript ist eine Open-Source Programmiersprache, die auf JavaScript basiert. Der Unterschied ist jedoch, dass bei TypeScript die Types der Variablen festgelegt werden müssen. Dies ermöglicht eine bessere Fehleranalyse und weniger Runtime-Fehlern. Zudem ist die Syntax quasi wie bei JavaScript, was wir alle bereits beherrschen. 

### Wieso kein JavaScript?
Wie bereits erwähnt, ist TypeScript Typesafe. Dies ermöglicht vorallem bei größeren Projekten mehr Konsistenz bei der Entwicklung und jeder sieht direkt, was für ein Type eine bestimmte Variable ist. Dies sind die wichtigsten Gründe, wieso wir uns für TypeScript entschieden haben, sowohl im Backend, als auch im Frontend wird auf TypeScript gesetzt. 


### Git/GitHub
Git ist ein Open-Source Versionskontrollsystem, das von GitHub verwendet wird. GitHub bietet also eine Plattform an, an der Entwickler gemeinsan an Projekten arbeiten kann und baut dabei auf Git auf, was die reine Versionskontrolle betrifft. Alternativen gibt es dazu keine wirklich erwähnenswerten, sodass wir uns für GitHub entschieden haben. Zudem wurde uns vorgeschrieben, alles auf GitHub zu posten ;)
